lib/access-context.js

```diff
@@ -126,9 +131,29 @@ AccessContext.prototype.addPrincipal = function(principalType, principalId, prin
 AccessContext.prototype.getUserId = function() {
   for (var i = 0; i < this.principals.length; i++) {
     var p = this.principals[i];
+    if (p.type === Principal.APP || p.type === Principal.ROLE || p.type == Principal.SCOPE) {
+      continue;
+    }
+    // the principalType must either be 'USER'
     if (p.type === Principal.USER) {
       return p.id;
     }
+    // or permit to resolve a valid user model
+    var userModel = this.registry.findModel(p.type);
+    if (!userModel) continue;
+    if (userModel.modelName === builtinUserModel.name) {
```

Comment:
yeah, we lost one of our discussion here: you commented:
> @bajtos : This does not work with a deeper hierarchy (Customer extends BaseUser extends built-in User).

to what i replied : 
> @ebarault : you're right, i noted that too. we should test recursively up to the top User model. I'll change the implem and add a test

i also asked in the same thread for a clean way to get the built-in User model name ("User") without hardcoding it